source/auth/tests/legacy/connection-string.yml

```diff
@@ -412,6 +412,50 @@ tests:
     mechanism_properties:
       ENVIRONMENT: azure
       TOKEN_RESOURCE: foo
+- description: should accept a url-encoded TOKEN_RESOURCE (MONGODB-OIDC)
+  uri: mongodb://user@localhost/?authMechanism=MONGODB-OIDC&authMechanismProperties=ENVIRONMENT:azure,TOKEN_RESOURCE:mongodb%3A%2F%2Ftest-cluster
+  valid: true
+  credential:
+    username: user
+    password: null
+    source: $external
+    mechanism: MONGODB-OIDC
+    mechanism_properties:
+      ENVIRONMENT: azure
+      TOKEN_RESOURCE: 'mongodb://test-cluster'
+- description: should accept an un-encoded TOKEN_RESOURCE (MONGODB-OIDC)
+  uri: mongodb://user@localhost/?authMechanism=MONGODB-OIDC&authMechanismProperties=ENVIRONMENT:azure,TOKEN_RESOURCE:mongodb://test-cluster
```

Comment:
Is this Uri valid? `TOKEN_RESOURCE:mongodb://test-cluster` Should we have `:` in the `mongodb://test-cluster` encoded as `%3A` (not sure about slash symbol though)?

Reply:
This usage is against the advice we give to users in the spec, but I think we should optimistically decode this (like [this in Java](https://github.com/mongodb/mongo-java-driver/pull/1371#discussion_r1572939075)), rather than failing. Although it is indeed ambiguous, I think we will never put a `:` in the name of any property (precisely because it is a separator).

Reply:
Yep, can do the same, just want to double check if we see this as expected behavior.